src/lib/battery/battery.cpp

```diff
@@ -223,17 +223,17 @@ Battery::determineWarning(bool connected)
 void
 Battery::computeScale()
 {
-	const float voltage_range = (_v_charged.get() - _v_empty.get());
+        // Using the filtered load voltage to give the most accurate reading.
+        // Checking that _voltage_filtered_v is non-zero
+        if(_voltage_filtered_v > 0.1f) {
+                const float scale = _v_charged.get() * _n_cells.get() / _voltage_filtered_v;
```

Comment:
Using the low passed voltage is problematic on current spikes. Did you do flight testing?

Reply:
Here is one flight, no auto-throttle (scale and throttle setting are scaled 10x):
![image](https://user-images.githubusercontent.com/5569656/50058877-749a6900-0187-11e9-8b10-29d921243a3d.png)

Here's another with autothrottle (scale and throttle setting are scaled 10x):
![image](https://user-images.githubusercontent.com/5569656/50059055-bed11980-018a-11e9-987c-bd5b305ed99f.png)

It is true that the filtered voltage ignores spikes and lags the actual voltage about 1-1,5 seconds. However:
- The unfiltered voltage data is noisy, and for the most part the filtered value follows the unfiltered voltage fairly well (tested with fixed wing trainer). I am not sure, but it could even end up amplifying the noise.
- I think that the scaling should rather be conservative (following the trend of the battery voltage) than aggressive. If there is a throttle spike lasting for 1 second that the scaling misses because of the lpf, I wouldn't consider that as a big miss. 

Unfortunately I don't have any flight testing data for multicopters and my test flights have been relaxed (not aggressive throttle pumping). But again it is just my opinion, of course there are other alternatives as well.